### PR TITLE
[RFC]: scripts/lint-commits: report github noreply emails

### PR DESCRIPTION
**Pull request by paper42**
ping @sgn

### CODE REVIEW

**Comment by Chocimier** [comment]
What's the point of asking for mail, when know github account of commiter?

**Comment by ericonr** [comment]
This is a big policy matter, because it means we won't accept contributions via GitHub UI anymore. Are we sure about it?

**Comment by paper42** [comment]
I sometimes contact package maintainers by email when I want to let them know about something (that's not suitable for an issue and they are not on IRC). I didn't have to contact a maintainer who doesn't have their email address public yet.

Also, people who have a github noreply email address often make their pull requests from the Github UI. I think that by itself is fine, but there is a higher probability that these changes are not well tested. Adding an email address shouldn't be a big problem for people. Void already requires email addresses in the maintainer field in package templates.

This is just a lint, so exceptions can be made, but I think having this information can be valuable.

> This is a big policy matter, because it means we won't accept contributions via GitHub UI anymore. Are we sure about it?

When people add a public email address to their account, they can use that address in the Github UI.

File: common/scripts/lint-commits [hunk 1]
```diff
@@ -31,6 +31,7 @@ for cmt in $("$GIT_CMD" rev-list --abbrev-commit $base..$tip)
 do
 	"$GIT_CMD" cat-file commit "$cmt" |
 	awk -vC="$cmt" '
+	(NR == 3) && /@users\.noreply\.github\.com>/ { print C ": commit author has a github noreply email" }
```
Comment (sgn) [inline common/scripts/lint-commits:34]: Notwithstanding other non-technical questions. `NR == 3` sounds incorrect.
In my opinion, it should be
```suggestion
	/^author .*noreply/ { print C ": commit author has a noreply email" }
```

Comment (paper42) [reply common/scripts/lint-commits]: Thanks, I modified your suggestion a bit to fix some potential false positives

File: common/scripts/lint-commits [hunk 2]
```diff
@@ -31,6 +31,7 @@ for cmt in $("$GIT_CMD" rev-list --abbrev-commit $base..$tip)
 do
 	"$GIT_CMD" cat-file commit "$cmt" |
 	awk -vC="$cmt" '
+	/author .*<.*noreply.*>/ { print C ": commit author has a noreply email" }
```
Comment (sgn) [inline common/scripts/lint-commits:34]: Please anchor `^author`

**Comment by paper42** [comment]
This proposal doesn't seem to have support among team members, closing.

**Pull request closed without merging**